http_opa/authorizer/authorizer.go

```diff
@@ -0,0 +1,252 @@
+		"application": a.application,
+	})
+
+	// Get the bearer token from the request
+	bearer, err := util.GetBearerFromRequest(req.(*http.Request))
+	if err != nil {
+		logger.WithError(err).Error("get_bearer_from_request")
+		return false, ctx, exception.ErrForbidden
+	}
+
+	// Verify the bearer token and get the raw JWT
+	claimsVerifier := a.claimsVerifier
+	if claimsVerifier == nil {
+		claimsVerifier = commonClaim.UnverifiedClaimFromBearers
+	}
+	rawJWT, errs := claimsVerifier([]string{bearer}, nil)
+	if len(errs) > 0 {
+		return false, ctx, exception.NewHttpError(
+			exception.WithError(multierr.Combine(errs...)),
+			exception.WithHttpStatus(http.StatusUnauthorized))
+	}
+
+	// Get the request ID from the request
+	reqID := util.GetRequestIdFromRequest(req.(*http.Request))
+
+	// Modify the endpoint if necessary
+	pargsEndpoint := endpoint
+	if a.endpointModifier != nil {
+		pargsEndpoint = a.endpointModifier.getModifiedEndpoint(pargsEndpoint)
+	}
+
+	// Create the Opa request payload
+	opaReq := Payload{
+		Endpoint:    pargsEndpoint,
+		FullMethod:  endpoint,
+		Application: a.application,
+		JWT:         common.RedactJWT(rawJWT),
+		RequestID:   reqID,
+		EntitledServices: a.entitledServices,
+	}
+
+	// Get the decision input for the request
+	decisionInput, err := a.decisionInputHandler.GetDecisionInput(ctx, endpoint, req)
+	if decisionInput == nil || err != nil {
+		logger.WithFields(log.Fields{
+			"endpoint": endpoint,
+		}).WithError(err).Error("get_decision_input")
+		return false, ctx, exception.ErrInvalidArg
+	}
+
+	opaReq.DecisionInput = *decisionInput
+
+	// Marshal the Opa request payload to JSON
+	opaReqJSON, err := json.Marshal(opaReq)
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"opaReq": opaReq,
+		}).WithError(err).Error("opa_request_json_marshal")
+		return false, ctx, exception.ErrInvalidArg
+	}
+
+	// Start a trace span for the Opa request
+	now := time.Now()
+	obfuscatedOpaReq := ShortenPayloadForDebug(opaReq)
+	logger.WithFields(log.Fields{
+		"opaReq": obfuscatedOpaReq,
+	}).Debug("opa_authorization_request")
+	ctx, span := trace.StartSpan(ctx, fmt.Sprint(SERVICENAME, endpoint))
+	{
+		span.Annotate([]trace.Attribute{
+			trace.StringAttribute("in", string(opaReqJSON)),
+		}, "in")
+	}
+
+	// Prepare the Opa input based on the decision document
+	var opaInput interface{}
+	opaInput = opaReq
+	if len(decisionInput.DecisionDocument) > 0 {
+		opaInput = OPARequest{Input: &opaReq}
+	}
+
+	// Execute the Opa evaluation
+	var opaResp OPAResponse
+	err = opaEvaluator(ctxlogrus.ToContext(ctx, logger), decisionInput.DecisionDocument, opaInput, &opaResp)
+	defer func() {
+		span.SetStatus(trace.Status{
+			Code:    int32(grpc.Code(err)),
+			Message: grpc.ErrorDesc(err),
+		})
+		span.End()
+		logger.WithFields(log.Fields{
+			"opaResp": opaResp,
+			"elapsed": time.Since(now),
+		}).Debug("authorization_result")
+	}()
+	if err != nil {
+		return false, ctx, err
+	}
```

Comment:
I would comment out these traces, and remove references to grpc. `http_opa` shouldn't require any version of grpc. Commenting out the traces will also remove the dependence on opentracing found in here. Maybe we should publish http_opa as a separate module too... so people aren't bound to the grpc version being imported in other parts of atlas-authz-middleware.

Reply:
Are we sure we want to remove the opentracing?  Wasn't there a requirement to add them?

Reply:
@cbarik-infoblox Please remove grpc related dependencies from http middleware. While we would need tracing for http; add a TODO and work towards getting the initial functionality.